searx/version.py

```diff
@@ -1,26 +1,173 @@
-# -*- coding: utf-8 -*-
-'''
-searx is free software: you can redistribute it and/or modify
-it under the terms of the GNU Affero General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
-
-searx is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU Affero General Public License for more details.
-
-You should have received a copy of the GNU Affero General Public License
-along with searx. If not, see < http://www.gnu.org/licenses/ >.
-
-(C) 2013- by Adam Tauber, <asciimoo@gmail.com>
-'''
-
-# version of searx
-VERSION_MAJOR = 1
-VERSION_MINOR = 0
-VERSION_BUILD = 0
-
-VERSION_STRING = "{0}.{1}.{2}".format(VERSION_MAJOR,
-                                      VERSION_MINOR,
-                                      VERSION_BUILD)
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+# pylint: disable=missing-function-docstring,missing-module-docstring,missing-class-docstring
+
+# some code is from https://github.com/pypa/setuptools_scm/blob/b0b46fd723f6649a7ba4895b005bc89db1529e22/src/setuptools_scm/utils.py  # pylint: disable=line-too-long
+# ( MIT License )
+
+import re
+import os
+import shlex
+import subprocess
+import logging
+
+
+logger = logging.getLogger('searx')
+
+
+class ShellException(Exception):
+    def __init__(self, command, exit_code, stdout, stderr):
+        super().__init__(self, stderr)
+        self.command = command
+        self.exit_code = exit_code
+        self.stdout = stdout
+        self.stderr = stderr
+
+
+def ensure_stripped_str(str_or_bytes):
+    if isinstance(str_or_bytes, str):
+        return str_or_bytes.strip()
+    return str_or_bytes.decode("utf-8", "surrogateescape").strip()
+
+
+def call_subprocess(cmd, cwd="."):
+    if os.name == "posix" and not isinstance(cmd, (list, tuple)):
+        cmd = shlex.split(cmd)
```

Comment:
sorry if I'm being annoying :-)  .. What use case (OS) do you have in mind for _non posix_ .. and what happens with the sub process call when ` os.name != "posix"`?

Reply:
My take was setup_scm has most probably dealt with various platform.

But, more in detail, from https://docs.python.org/3/library/subprocess.html#subprocess.Popen :

> On Windows, if args is a sequence, it will be converted to a string in a manner described in Converting an argument sequence to a string on Windows. This is because the underlying CreateProcess() operates on strings.

And according to https://stackoverflow.com/questions/1854/python-what-os-am-i-running-on `os.name != 'posix'` only on Windows (and Jython which is out of scope and perhaps some other implementation I'm not aware of, yes it should be `os.name != 'nt' according to what I understand even in setup_scm).

But anyway `searx/version.py` works on Windows even with the commit 7710c9422fceda1d45cbdda634b80b89ceee2ac6